templates/csharp/custom-copilot-weather-agent/Program.cs.tpl

```diff
@@ -45,11 +40,20 @@ builder.Services.AddTransient<WeatherForecastAgent>();
 // Add AspNet token validation
 builder.Services.AddBotAspNetAuthentication(builder.Configuration);
 
+// Register IStorage.  For development, MemoryStorage is suitable.
+// For production Agents, persisted storage should be used so
+// that state survives Agent restarts, and operate correctly
+// in a cluster of Agent instances.
+builder.Services.AddSingleton<IStorage, MemoryStorage>();
+
+// Add AgentApplicationOptions from config.
+builder.AddAgentApplicationOptions();
+
 // Add AgentApplicationOptions.  This will use DI'd services and IConfiguration for construction.
 builder.Services.AddTransient<AgentApplicationOptions>();
 
 // Add the bot (which is transient)
-builder.AddBot<{{SafeProjectName}}.Bot.WeatherAgentBot>();
+builder.AddAgent<TEST.Bot.WeatherAgentBot>();
```

Comment:
Must this be `TEST.Bot.WeatherAgentBot`?

Reply:
I'll fix this in next PR.